### ISSUE TITLE
Angular Material conflicts with Angular Google Maps once markers are added to map. 

### ISSUE THREAD

**Issue by heinschulie** [angular/material#4004]
Hi everybody, 

My issue is so convoluted that I'm scared the solution is really obvious...I recently rehauled the design of a small app I am writing which relies heavily on Angular Google Maps. The new look uses the Angular Material framework and everything works amazingly well on Safari but for some reason on Chrome the map breaks completely (jumps erratically when attempting to pan etc.) as soon as markers are added to it. 

Has anyone had any similar experiences working with these two frameworks? Or just any experience that may be of help? I have rebuilt the app up twice now trying to pin point the cause and just can't seem to find it.    

I realize this involves conflict between two frameworks but I think quite a few people are going to be wanting to take advantage of both so hopefully this question helps more people than just me. I will cross-post it on the Angular Google Map forums too and update both with any useful information. 

Thanks so much in advance and let me know if more context is needed! 
Hein 

My code is pretty generic but I'll include snippets below: 

HTML

```html
<div layout="row" flex>
    <ui-gmap-google-map center='map.center'
                        zoom='map.zoom'
                        pan="true"
                        options="map.options"
                        events="map.events"
                        control="map.control"
                        bounds="map.bounds"
                        refresh="city">

        <ui-gmap-marker idkey='userState.data.user.id'
                        coords='userState.data.user.position'
                        options="userState.data.user.options">
        </ui-gmap-marker>

    </ui-gmap-google-map>
</div>
```

CONTROLLER

```js
'use strict';
var app = angular.module('myApp');
app.controller('mapController', ["$scope", "$timeout", "$log", "geoLocationService", "uiGmapGoogleMapApi", "userState", function ($scope, $timeout, $log, geoLocationService, uiGmapGoogleMapApi, userState) {

    $scope.serviceBase = appService.serviceBase;
    $scope.userState = userState;

    uiGmapGoogleMapApi.then(function (maps) {

         // ********************************************************************** INITIALIZATION AND MAP CALIBRATION AND METHODS //

        var styleArray = [
              {
                  featureType: "all",
                  stylers: [
                    { saturation: -80 }
                  ]
              }, {
                  featureType: "road.arterial",
                  elementType: "geometry",
                  stylers: [
                    { hue: "#00ffee" },
                    { saturation: 50 }
                  ]
              }, {
                  featureType: "poi.business",
                  elementType: "labels",
                  stylers: [
                    { visibility: "off" }
                  ]
              }
        ];
        var mapOptions = {
            styles: styleArray,
            disableDefaultUI: true
        }

        // *** EVENTS *** // 

        function boundsChanged() {
            //If the city isn't set - this is the first time this map is instantiated. Try return to initial promise chain 
            if (!$scope.city)
                return;

            $scope.$apply(function () {
                var bounds = $scope.map.bounds;
                var center = $scope.map.center;

                // Bounds might not always be set so check that it exists first.
                if (bounds.northeast && center.latitude) {
                    var lat1 = center.latitude;
                    var lng1 = center.longitude;
                    var lat2 = bounds.northeast.latitude;
                    var lng2 = bounds.northeast.longitude;
                    $scope.mapPosition = [];
                    $scope.mapPosition.push(lat1);
                    $scope.mapPosition.push(lng1);
                    $scope.mapRadius = geoLocationService.haversineDistance(lat1, lng1, lat2, lng2);
                    $scope.mapPosition.push($scope.mapRadius);
                }

        $scope.map = {
            center: { latitude: 45, longitude: -73 },
            zoom: 14,
            pan: true,
            options: mapOptions,
            control: {},
            markersControl: {},
            events: { 'idle': boundsChanged },
            bounds: {},
            showDirections: false
        };

        //INITIALIZATION : Find location of user

        var findLatLong = function () {
            return $scope.userState.callForUserLocation($scope, false)
               .then(function (position) {
                   $scope.map.center.latitude = position.coords.latitude;
                   $scope.map.center.longitude = position.coords.longitude;
                   var latLong = position.coords.latitude + "," + position.coords.longitude;
                   return latLong;
               });
        };

        var reverseGeoCodeCity = function (coordinates) {
            return geoLocationService.reverseGeoCodeCity($scope, coordinates, "app-map-canvas").then(function (results) {
                $scope.city = results;
                return coordinates;
            });
        };

        $scope.userState.callForUser()
            .then(findLatLong)
            .then(reverseGeoCodeCity);

}]);
```

**Comment by topherfangio** [comment]
Any chance you have a test site that we could look at, or could you possibly create a Codepen.io with a minimal, breaking example.

Also, what version of Angular/Material/Maps/Chrome are you using?

**Comment by heinschulie** [comment]
Hi Topher, 

Firstly thanks for the reply! 

It's relatively late here and I've shut down everything but will try get a code pen up tmw morning. I'll include the exact versions of everything then too. 

Thanks again! 
Hein

**Comment by heinschulie** [comment]
Hi @topherfangio 

Apologies for not putting this up sooner. This is the first time I ever post an issue or write a plunkr for public viewing. Let me know if anything else is needed! 

If anyone wishes to follow the other thread on this issue over on the Maps forum, here is the link: https://github.com/angular-ui/angular-google-maps/issues/1420

Else here is a plunk of the issue: http://plnkr.co/edit/QHmv0yRJS396hLcPJHKs
And here is a running demo site: http://matmapchromeissue.azurewebsites.net/

Some more context: This issue only appears on the mobile version of Chrome, not the desktop version nor the emulator view.  Furthermore, I've tested it now on an iPhone 4, 5, 6 and HTC 1 - the bug appears on all the iPhone Chromes but not on the HTC1. I have had reports though of other android users struggling - so I don't believe it is limited to iOs devices. 

To make things more complicated - my plunkr runs fine through Chrome, hence I stood up a quick site for anyone to see the exact effect of what I'm talking about. The EXACT same code is running on both. 

Framework versions: 
Angular : 1.3.15
Material: 0.9.4
Angular-Google-Maps: 2.1.5
Chrome versions vary. 

Thanks! 
Hein

**Comment by topherfangio** [comment]
@heinschulie I hate to say it, but I'm afraid I can't reproduce the issue on my iPhone 6. I don't have much else to test with, but since I can't even see what the issue is, I don't think I can really help debug it.

Is this a bug that happens 100% consistently, or is it something that only pops up occasionally?

Finally, I noticed you're using Material 0.9.4. Is there any chance you could test 0.10.0 and also the latest master versions to see if that fixes the issue?

**Comment by heinschulie** [comment]
Hi @topherfangio , 

Man, that is either great news (the bug has vanished) or just damn frustrating! My guess is it's the latter. My experience with it is that on the real-world site, where the data behind the markers is slightly heavier, the bug happens consistently and immediately... on the test site I mocked up it happens around the 20th time the idle event fires, that is to say - when there are around 20 markers on the map. 

I have tried using the newer versions of both frameworks but will update the plunkr and test site as you advise. 

The chap helping out on this issue on the maps forum noticed that the event actually fires twice and that debouncing may help with the issue - but what is actually REALLY confusing is that both these frameworks work amazingly well in isolation... but that Material causes the map to bomb out under certain conditions. 

If you do perhaps have access to an iPhone5 or 4 with Chrome it would be great to know that you have at least seen the bug described - if only to ascertain whether I'm going crazy or not! 

Thanks again for your help.
